x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/util/PlanStreamOutput.java

```diff
@@ -21,4 +24,13 @@ public interface PlanStreamOutput {
      * @throws IOException
      */
     boolean writeAttributeCacheHeader(Attribute attribute) throws IOException;
+
+    static void writeEsField(PlanStreamOutput output, EsField field) throws IOException {
+        output.writeEsField(field);
+    }
+
+    void writeEsField(EsField field) throws IOException;
```

Comment:
It'd be nice if these could live in `EsField` and it did the cache header dance there, I think.

Reply:
The cache logic will have to be in PlanStreamOutput/Input anyway, but I think I make EsField the API entry point

Reply:
Ok, write logic can easily be migrated to EsField.
Read logic is a bit more intricate, but at least I can make the APIs consistent with `EsField` `writeTo` and `readFrom`